# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerService.h
# Commit: 01fe98c08d10
# Full Hash: 01fe98c08d102c28e3a61209ea31834eff83202a
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-25 21:02:33
# Regressor Bug: 1899503
# File Overlap Count: 4
# Description:
#   Bug 1899503 - P3 PRemoteWorkerDebugger binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebugger binding.
#   
#   We are trying to keep the same logic debugger registration/unregistration to minimize the difference between the local debugger(the debugger on the content process main thread)
# ==============================================================================

diff -r 9bd2a82481f4 -r 01fe98c08d10 dom/workers/remoteworkers/RemoteWorkerService.h
--- a/dom/workers/remoteworkers/RemoteWorkerService.h	Tue Mar 25 09:13:20 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerService.h	Tue Mar 25 09:13:20 2025 +0000
@@ -13,12 +13,14 @@
 #include "nsCOMPtr.h"
 #include "nsIObserver.h"
 #include "nsISupportsImpl.h"
+#include "mozilla/dom/RemoteWorkerTypes.h"
 
 class nsIThread;
 
 namespace mozilla::dom {
 
 class RemoteWorkerDebuggerManagerChild;
+class RemoteWorkerDebuggerManagerParent;
 class RemoteWorkerService;
 class RemoteWorkerServiceChild;
 class RemoteWorkerServiceShutdownBlocker;
@@ -123,7 +125,8 @@
 
   nsCOMPtr<nsIThread> mThread;
   RefPtr<RemoteWorkerServiceChild> mActor;
-  RefPtr<RemoteWorkerDebuggerManagerChild> mDebuggerManagerActor;
+  RefPtr<RemoteWorkerDebuggerManagerChild> mDebuggerManagerChild;
+  RefPtr<RemoteWorkerDebuggerManagerParent> mDebuggerManagerParent;
   // The keep-alive is set and cleared on the main thread but we will hand out
   // additional references to it from the "Worker Launcher" thread, so it's
   // appropriate to use a mutex.  (Alternately we could have used a ThreadBound